lib/postprocess/processImage.js

```diff
@@ -107,11 +119,11 @@ module.exports = async function processImage(files, drawingEl, newRelIdCounterMa
       'base64'
     )
   } else {
-    const response = await axios({
+    const response = await limit(() => axios({
```

Comment:
**note:** this has been built to limit the download of images to 64 in parallel. it depends on whether the underlying architecture of` jsreport-docx` has been changed or not whether this still makes sense though. the root incentive here was that jsreport runs on a node.js server and not is not super efficient running a high amount of downloads in parallel (we had customers export thousands of tickets which tens of images)

Reply:
@sebastian-schlecht The latest version of the plugin contains configuration for [imageFetchParallelLimit](https://github.com/jsreport/jsreport/blob/master/packages/jsreport-docx/jsreport.config.js) I configured that property to 64 as well in [report-service-v2 PR](https://github.com/capmo/report-service-v2/pull/27/files#diff-5b8329d7cc0f10d3138211c14c7b9a211b9f5e9b17412c1117cd94a35536a699)
I think it's fine to leave this p-limit code usage here as well. The underlying architecture is not changed at all. They download the images as in-memory buffer and use them, so not very efficient with maaany images.

Reply:
I did not know that [this](https://github.com/capmo/report-service-v2/pull/27/files#diff-5b8329d7cc0f10d3138211c14c7b9a211b9f5e9b17412c1117cd94a35536a699R5) configuration was possible, I think it wasn't there when I built this. If this is the case, we could even leave it out given that the process is still efficient enough to render 1000k pages in word. A simple load test using artificial data could do here.

Reply:
Fair enough. I think however that when jsreport runs in multi-process mode (I remember there was sth like that where it spawns a node process per rendering request), then this should be equivalent.